libtenzir/include/tenzir/exec/checkpoint.hpp

```diff
@@ -0,0 +1,19 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+namespace tenzir::exec {
+
+struct checkpoint {
```

Comment:
I believe for features like load-balancing and split-routing we need the checkpoint to have shared-resource semantics.
For example if we have `from_kafka | load_balance ...` we need to insert the checkpoint into each subpipeline and wait for all copies to be commited until we can update the read offset the post-commit action in `from_kafka`.

Reply:
This is already happening. The `load_balance` operator has a subpipeline that gets started multiple times. The subpipeline each forward the checkpoints, which `load_balance` itself collects, and then once it received checkpoints from all its pipelines it forwards them in stream to the executor.